front/components/vaults/VaultResourcesList.tsx

```diff
@@ -155,13 +157,90 @@ export const VaultResourcesList = ({
     );
   }
 
-  const setUpDataSources = vaultDataSourceViews.map((dsv) => dsv.connectorId);
-  const unusedDataSources = managedDataSources.filter(
-    (ds) => !setUpDataSources.includes(ds.connectorId)
-  );
+  const saveSelectedDatasources = async (
+    selectedDataSources: VaultSelectedDataSources
+  ) => {
+    let error = null;
+    await Promise.all(
+      selectedDataSources.map(async (sDs) => {
```

Comment:
shouldn't we delete the views that are not selected anymore ?

Reply:
Yes that's what I said in the main post: 
> Then I need to update the patch endpoint to remove a datasource view if the content is empty, cause if you unselect everything in a Notion it will keep the Notion line and if we click it says "no content".

Happy to do it in this PR if you prefer, otherwise this will be my next PR

Reply:
Done in this PR!